testservice_staticroutes.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2012-2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package gomaasapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"regexp"
+)
+
+func getStaticRoutesEndpoint(version string) string {
+	return fmt.Sprintf("/api/%s/static-routes/", version)
+}
+
+// TestStaticRoute is the MAAS API Static Route representation
+type TestStaticRoute struct {
+	Destination TestSubnet `json:"destination"`
+	Source      TestSubnet `json:"source"`
+	Metric      uint       `json:"metric"`
+	GatewayIP   string     `json:"gateway_ip"`
+	ResourceURI string     `json:"resource_uri"`
+	ID          uint       `json:"id"`
+	// These are internal bookkeeping, and not part of the public API, so
+	// should not be in the JSON
+	sourceCIDR      string `json:"-"`
+	destinationCIDR string `json:"-"`
+}
+
+// staticRoutesHandler handles requests for '/api/<version>/static-routes/'.
+func staticRoutesHandler(server *TestServer, w http.ResponseWriter, r *http.Request) {
+	values, err := url.ParseQuery(r.URL.RawQuery)
+	checkError(err)
+	op := values.Get("op")
+	if op != "" {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	staticRoutesURLRE := regexp.MustCompile(`/static-routes/(.+?)/`)
+	staticRoutesURLMatch := staticRoutesURLRE.FindStringSubmatch(r.URL.Path)
+	staticRoutesURL := getStaticRoutesEndpoint(server.version)
+
+	var ID uint
+	var gotID bool
+	if staticRoutesURLMatch != nil {
+		// We pass a nil mapping, as static routes don't have names, but this gives
+		// consistent integers and range checking.
+		ID, err = NameOrIDToID(staticRoutesURLMatch[1], nil, 1, uint(len(server.staticRoutes)))
+
+		if err != nil {
+			http.NotFoundHandler().ServeHTTP(w, r)
+			return
+		}
+
+		gotID = true
+	}
+
+	switch r.Method {
+	case "GET":
+		w.Header().Set("Content-Type", "application/vnd.api+json")
+		if len(server.staticRoutes) == 0 {
+			// Until a static-route is created, behave as if the endpoint
+			// does not exist. This way we can simulate older MAAS
+			// servers that do not support spaces.
+			http.NotFoundHandler().ServeHTTP(w, r)
+			return
+		}
+
+		if r.URL.Path == staticRoutesURL {
+			var routes []*TestStaticRoute
+			// Iterating by id rather than a dictionary iteration
+			// preserves the order of the routes in the result.
+			for i := uint(1); i < server.nextStaticRoute; i++ {
+				route, ok := server.staticRoutes[i]
+				if ok {
+					server.setSubnetsOnStaticRoute(route)
+					routes = append(routes, route)
+				}
+			}
+			err = json.NewEncoder(w).Encode(routes)
+		} else if gotID == false {
+			w.WriteHeader(http.StatusBadRequest)
+		} else {
+			err = json.NewEncoder(w).Encode(server.staticRoutes[ID])
+		}
+		checkError(err)
+	case "POST":
```

Comment:
I was copying this, but I went ahead and put in at least http.StatusNotImplemented.